render-ws-spark-client/src/main/java/org/janelia/render/client/spark/match/ClusterCountClient.java

```diff
@@ -46,63 +48,80 @@ public org.janelia.render.client.ClusterCountClient buildJavaClient() {
             javaClientParameters.tileCluster = tileCluster;
             return new org.janelia.render.client.ClusterCountClient(javaClientParameters);
         }
-
-        /** @return client specific parameters populated from specified alignment pipeline parameters. */
-        public static Parameters fromPipeline(final AlignmentPipelineParameters alignmentPipelineParameters) {
-            final Parameters derivedParameters = new Parameters();
-            derivedParameters.multiProject = alignmentPipelineParameters.getMultiProject();
-            derivedParameters.tileCluster = alignmentPipelineParameters.getTileCluster();
-            return derivedParameters;
-        }
-
     }
 
+    /** Run the client with command line parameters. */
     public static void main(final String[] args) {
-
         final ClientRunner clientRunner = new ClientRunner(args) {
             @Override
             public void runClient(final String[] args) throws Exception {
-
                 final Parameters parameters = new Parameters();
                 parameters.parse(args);
-
-                final ClusterCountClient client = new ClusterCountClient(parameters);
-                client.run();
-
+                final ClusterCountClient client = new ClusterCountClient();
+                client.createContextAndRun(parameters);
             }
         };
         clientRunner.run();
+    }
 
+    /** Empty constructor required for alignment pipeline steps. */
+    public ClusterCountClient() {
     }
 
-    private final Parameters parameters;
+    /** Create a spark context and run the client with the specified parameters. */
+    public void createContextAndRun(final Parameters clientParameters) throws IOException {
+        final SparkConf conf = new SparkConf().setAppName(getClass().getSimpleName());
+        try (final JavaSparkContext sparkContext = new JavaSparkContext(conf)) {
+            LOG.info("run: appId is {}", sparkContext.getConf().getAppId());
+            findConnectedClusters(sparkContext, clientParameters);
+        }
+    }
 
-    public ClusterCountClient(final Parameters parameters) throws IllegalArgumentException {
-        LOG.info("init: parameters={}", parameters);
-        this.parameters = parameters;
+    /** Validates the specified pipeline parameters are sufficient. */
+    @Override
+    public void validatePipelineParameters(final AlignmentPipelineParameters pipelineParameters)
+            throws IllegalArgumentException {
+        AlignmentPipelineParameters.validateRequiredElementExists("tileCluster",
+                                                                  pipelineParameters.getTileCluster());
     }
 
-    public void run() throws IOException {
+    /** Run the client as part of an alignment pipeline. */
+    public void runPipelineStep(final JavaSparkContext sparkContext,
+                                final AlignmentPipelineParameters pipelineParameters)
```

Comment:
I really like how the changed handling of the parameter object now makes the steps seem like proper pipeline steps (the parameters now act as shared context instead of internal state of the steps).